assets/src/edit-story/app/userOnboarding/useUserOnboarding.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+/**
+ * Internal dependencies
+ */
+// import { useCurrentUser } from '../currentUser';
+// import { useHelpCenter } from '../helpCenter';
+
+const TRIGGER_CONTEXTS = {
+  SAFE_ZONE: 'safeZone',
+  // todo [@embarks]: other onboarding triggers
+};
```

Comment:
I'm wondering if these triggers are different than just using the tips data constant from the help center?

Will these triggers always map 1:1 to opening up a single tip in the help center? or are there other scenarios that wouldn't be so 1:1 and necessitate extra data that maps to the tip data?

Reply:
This is a great question. What I was thinking was we might have onboarding steps that are not related to the help center. We might want to trigger a highlight, or some other step-based onboarding, or track the items separately. But I might be pre-optimizing and I think if all they do is open a tip we should just use the same constants.

But I think I need to open up a discussion about what the UX should be around these triggers. Right now it's variable what in-context triggers open which tip. For masking, for example, the ratio is 4:1 for in-context-triggers to tips. So I need to know if we need to track an in-context trigger event independently from whether the user has read the tip.

Reply:
Oh yea, didn't think about having other actions than opening the help center in here. good call!